packages/flutter_tools/lib/src/commands/attach.dart

```diff
@@ -399,7 +443,32 @@ class MDnsObservatoryPortDiscovery {
         printError('Unexpectedly found more than one observatory report for $domainName '
                    '- using first one (${srv.first.port}).');
       }
-      return srv.first.port;
+      printStatus('Checking for authentication code for $domainName');
+      final List<TxtResourceRecord> txt = await client
+        .lookup<TxtResourceRecord>(
+            ResourceRecordQuery.text(domainName),
+        )
+        .toList();
+      String authCode = '';
```

Comment:
How about:

```
if (txt.isEmpty) {
  return MDnsObservatoryDiscoveryResult(srv.first.port, '');
}
```

Then unnest. There are other opportunities to flip branches and unnest below as well.